definitions/marts/looker_studio_marts/ecf/ls_ecf_mentor_backfill_grant_funding.sqlx

```diff
@@ -0,0 +1,212 @@
+   }
+}
+
+WITH declarations AS (
+  SELECT
+     dec.declaration_id
+    ,dec.participant_profile_id
+    ,dec.user_id
+    ,dec.TRN
+    ,dec.declaration_type
+    ,dec.state AS declaration_state
+    ,CASE
+      WHEN dec.declaration_type = 'started' THEN 1
+      WHEN dec.declaration_type = 'retained-1' THEN 2
+      WHEN dec.declaration_type = 'retained-2' THEN 3
+      WHEN dec.declaration_type = 'retained-3' THEN 4
+      WHEN dec.declaration_type = 'retained-4' THEN 5
+      WHEN dec.declaration_type = 'extended-1' THEN 6
+      WHEN dec.declaration_type = 'extended-2' THEN 7
+      WHEN dec.declaration_type = 'extended-3' THEN 8
+      WHEN dec.declaration_type = 'completed' THEN 9
+      ELSE NULL
+     END AS declaration_type_hierarchy
+    ,DATE(dec.declaration_date) AS declaration_date
+    ,dec.cpd_lead_provider_name AS lead_provider_name
+    ,COALESCE(dp.cpd_dp_name, dec.delivery_partner_name) AS delivery_partner_name
+    ,dec.induction_record_id AS latest_induction_record_id
+    ,dec.school_urn AS induction_record_school_urn -- School URN of latest induction record.    
+    ,COALESCE(ind.school_urn, ind2.school_urn) AS declaration_date_school_urn -- School URN on the induction record valid on the declaration date.
+    ,COALESCE(ind.induction_record_id, ind2.induction_record_id) AS induction_record_id_on_declaration_date
+    ,COALESCE(ind.school_urn, ind2.school_urn, dec.school_urn) AS used_school_urn
+    -- For each started, retained-1/2/3/4 or completed declaration 6 hours of funding is allocated.
+    ,CASE
+      WHEN dec.declaration_type IN ('started', 'retained-1', 'retained-2', 'retained-3', 'retained-4', 'completed') THEN 6
+      ELSE 0
+     END AS funded_hours
+    ,CASE
+      WHEN ROW_NUMBER() OVER(PARTITION BY COALESCE(ind.school_urn, ind2.school_urn, dec.school_urn)) = 1 THEN 1
+      ELSE 0
+     END AS unique_school_urn_count
+    ,CASE 
+      WHEN (ROW_NUMBER() OVER(PARTITION BY dec.declaration_id ORDER BY COALESCE(ind.start_date, ind2.start_date) DESC)) = 1 THEN 1 
+      ELSE 0 
+     END AS rn0
+  FROM 
+    ${ref('ecf_declarations')} AS dec
+  LEFT JOIN
+    ${ref('ecf_inductions')} AS ind
+  ON
+    ind.participant_profile_id = dec.participant_profile_id
+    AND
+    DATE(dec.declaration_date) BETWEEN DATE(ind.start_date) AND DATE(IFNULL(ind.end_date, '2050-12-31'))
+        
+  LEFT JOIN
+    ${ref('ecf_inductions')} ind2
+  ON
+    ind2.participant_profile_id = dec.participant_profile_id
+    AND
+    DATE(dec.declaration_date) < IFNULL(DATE(ind2.end_date), '2050-12-31')
+    AND
+    ind.induction_record_id IS NULL
+
+  LEFT JOIN
+    ${ref('ls_declarations_provider_names')} AS dp
+  ON
+    dp.id = dec.declaration_id
+
+  WHERE
+    funded_declaration = TRUE
+    AND
+    participant_course = 'Mentor'
+    AND
+    dec.declaration_date >= '2023-09-01'
+  QUALIFY (
+    rn0 = 1
+  )
+),
+
+ects AS (

```

Comment:
I think all of this could come from ecf_inductions_dedupe? Participant profiles coverage of school ids is patchy, we shouldn't use it for that.